src/components/StudentProposals.tsx

```diff
@@ -1,27 +1,72 @@
-import { H2, H3 } from '@uzh-bf/design-system'
-import { RefObject } from 'react'
+import { H2, H3, Select } from '@uzh-bf/design-system'
+import * as R from 'ramda'
+import { RefObject, useMemo } from 'react'
+import { ProposalDetails, ProposalStatusFilter } from 'src/types/app'
 import ProposalCard from './ProposalCard'
 
 interface StudentProposalsProps {
-  data: any
-  groupedStudentProposals: any
+  data: ProposalDetails[]
   selectedProposal: string | null
   setSelectedProposal: (proposalId: string | null) => void
-  setDisplayMode: (displayMode: string) => void
   buttonRef: RefObject<HTMLButtonElement>
+  filters: {
+    status: ProposalStatusFilter
+  }
+  setFilters: (filters: { status: ProposalStatusFilter }) => void
 }
 
 export default function StudentProposals({
   data,
-  groupedStudentProposals,
   selectedProposal,
   setSelectedProposal,
-  setDisplayMode,
   buttonRef,
+  filters,
+  setFilters,
 }: StudentProposalsProps) {
+  const groupedStudentProposals = useMemo(() => {
+    if (!data) return {}
+
+    return R.groupBy<ProposalDetails>(
+      (p) => p.topicArea.name,
+      R.sortBy(
+        R.prop('title'),
+        data.filter(
+          (proposal: ProposalDetails) => proposal.typeKey === 'STUDENT'
+        )
+      )
+    )
+  }, [data])
+
   return (
     <div>
-      <H2>Student Proposals</H2>
+      <div className="flex items-center justify-between">
+        <H2>Student Proposals</H2>
+        <Select
+          value={filters.status}
+          items={[
+            {
+              value: ProposalStatusFilter.OPEN_PROPOSALS,
```

Comment:
reorder the items 1. open, 2. my proposals, 3. rejected/declined, 4. all proposals

Reply:
Done